internal/storage/s3/s3.go

```diff
@@ -100,10 +111,21 @@ func (v *s3Storage) Name() string {
 // Copy copies the given file to the S3/Minio storage backend.
 func (b *s3Storage) Copy(file string) error {
 	_, name := path.Split(file)
+	srcFileInfo, err := os.Stat(file)
+
+	if err != nil {
+		return fmt.Errorf("(*s3Storage).Copy: error reading the local file: %w", err)
+	}
+
+	_, partSize, _, err := minio.OptimalPartInfo(srcFileInfo.Size(), uint64(b.internal.defaultPartSize*1024*1024))
+	if err != nil {
+		return fmt.Errorf("(*s3Storage).Copy: error computing the optimal s3 part size: %w", err)
+	}
```

Comment:
I would think it'd be great to run all of this code only when a non-default value for `defaultPartSize` is used.